### PR TITLE
Add a new group in the IODA format for L2 products with averaging kernel and slant columns

### PR DESCRIPTION
**Pull request by jeromebarre**
This PR moves all the quantities required for trace gases L2 retrieval smoothing in UFO that were in `Metadata` to a new group called `Smoothing`.
This group should encompasses all the quantities needed to perform L2 trace gases specific transformation:
- averaging kernel
- apriori
- airmass factors
- etc...

I also took the opportunity to do some cleaning. The avk precision would never be used in UFO as far as I can conceptualise UFO...

This impacts ufo, ufo-data and fv3-jedi-data
Link PRs below:

https://github.com/JCSDA-internal/ufo/pull/2183
https://github.com/JCSDA-internal/ufo-data/pull/199
https://github.com/JCSDA-internal/fv3-jedi-data/pull/41

Issue related to this PR:
https://github.com/JCSDA-internal/ioda-converters/issues/787

### CODE REVIEW

File: src/compo/mopitt_co_nc2ioda.py [hunk 1]
```diff
@@ -158,11 +158,11 @@ def _read(self):
                 self.outdata[('datetime', 'MetaData')] = times[flg]
                 self.outdata[('latitude', 'MetaData')] = lats[flg]
                 self.outdata[('longitude', 'MetaData')] = lons[flg]
-                self.outdata[('apriori_term', 'MetaData')] = ap_tc[flg]
+                self.outdata[('apriori_term', 'Smoothing')] = ap_tc[flg]
```
Comment (CoryMartin-NOAA) [inline src/compo/mopitt_co_nc2ioda.py:161]: Said this in the UFO PR, not a huge fan of `Smoothing` (is this added to the IODA conventions?). What about something more like `Retrieval`? But, I don't feel strong enough to derail this/associated PRs.

**Review by CoryMartin-NOAA — APPROVED** [review]

File: src/compo/mopitt_co_nc2ioda.py (hunk 1 above)
Comment (karpob) [reply src/compo/mopitt_co_nc2ioda.py:161]: I'm kinda with you on this. Something with `Retrieval` in it like `RetrievalAncillaryData` or something, but that just seems bloated and awkward. I don't have a good alternative either. Also, don't feel strongly enough to derail things.

**Review by karpob — APPROVED** [review]

File: src/compo/mopitt_co_nc2ioda.py (hunk 1 above)
Comment (jeromebarre) [reply src/compo/mopitt_co_nc2ioda.py:161]: See comment response here: https://github.com/JCSDA-internal/ufo/pull/2183#discussion_r937136336
I like Bryan's second suggestion which is the most general for all the data we want to put in this group. I could change it to `RtrvlAncData` ?

Comment (karpob) [reply src/compo/mopitt_co_nc2ioda.py:161]: Sure, works for me.

Comment (jeromebarre) [reply src/compo/mopitt_co_nc2ioda.py]: I changed the name of the group to something less awkward now.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**